scripts/count-users-entered-venue.ts

```diff
@@ -1,33 +1,41 @@
-#!/usr/bin/env node --experimental-json-modules --loader ts-node/esm
+#!/usr/bin/env node -r esm -r ts-node/register
 
-import admin from "firebase-admin";
-
-import { initFirebaseAdminApp } from "./lib/helpers.js";
+import { resolve } from "path";
 
-const usage = () => {
-  const scriptName = process.argv[1];
-  const helpText = `
----------------------------------------------------------  
-${scriptName}: Count the users that have entered the venue(s) specified
+import admin from "firebase-admin";
 
-Usage: node ${scriptName} PROJECT_ID VENUE_IDS
+import { initFirebaseAdminApp, makeScriptUsage } from "./lib/helpers";
 
-Example: node ${scriptName} co-reality-map venueId,venueId2,venueIdN
----------------------------------------------------------
-`;
+const usage = makeScriptUsage({
+  description: "Count the users that have entered the venue(s) specified.",
+  usageParams: "PROJECT_ID VENUE_IDS [CREDENTIAL_PATH]",
+  exampleParams:
+    "co-reality-map venueId,venueId2,venueIdN [theMatchingAccountServiceKey.json]",
+});
 
-  console.log(helpText);
-  process.exit(1);
-};
+const [projectId, venueIds, credentialPath] = process.argv.slice(2);
 
-const [projectId, venueIds] = process.argv.slice(2);
+// Note: no need to check credentialPath here as initFirebaseAdmin defaults it when undefined
 if (!projectId || !venueIds) {
   usage();
 }
 
 const venueIdsArray = venueIds.split(",");
 
-initFirebaseAdminApp(projectId);
+// Note: if we ever need to handle this, we can split our firestore query into 'chunks', each with 10 items per array-contains-any
+if (venueIdsArray.length > 10) {
+  console.error(
+    "Error: This script can only handle up to 10 venueIds at once at the moment."
```

Comment:
Why? Is this a firebase restriction or just our own limitation to avoid heavy performance issues.

Reply:
Firebase restriction in that we can only pass 10 items to the ‘array contains’ things.

We can work around it with the ‘chunk’ pattern (as explained in the attached comment above the check)

I’ve never had to put that many venueId’s into this sort of script, and so didn’t want to spend the extra time/script complexity of implementing the chunk pattern for a need that may not ever arise.